x-pack/examples/embedded_lens_example/public/app.tsx

```diff
@@ -0,0 +1,172 @@
+                },
+              },
+            },
+          },
+        },
+      },
+      filters: [],
+      query: { language: 'kuery', query: '' },
+      visualization: {
+        axisTitlesVisibilitySettings: { x: true, yLeft: true, yRight: true },
+        fittingFunction: 'None',
+        gridlinesVisibilitySettings: { x: true, yLeft: true, yRight: true },
+        layers: [
+          {
+            accessors: ['col2'],
+            layerId: 'layer1',
+            seriesType: 'bar_stacked',
+            xAccessor: 'col1',
+            yConfig: [{ forAccessor: 'col2', color }],
+          },
+        ],
+        legend: { isVisible: true, position: 'right' },
+        preferredSeriesType: 'bar_stacked',
+        tickLabelsVisibilitySettings: { x: true, yLeft: true, yRight: true },
+        valueLabels: 'hide',
+      },
+    },
+  };
+}
+
+export const App = (props: { core: CoreStart; plugins: StartDependencies }) => {
+  const [color, setColor] = useState('green');
+  const LensComponent = props.plugins.lens.EmbeddableComponent;
+  return (
+    <EuiPage>
+      <EuiPageBody style={{ maxWidth: 1200, margin: '0 auto' }}>
+        <EuiPageHeader>
+          <EuiPageHeaderSection>
+            <EuiTitle size="l">
+              <h1>Embedded Lens vis</h1>
+            </EuiTitle>
+          </EuiPageHeaderSection>
+        </EuiPageHeader>
+        <EuiPageContent>
+          <EuiPageContentBody style={{ maxWidth: 800, margin: '0 auto' }}>
+            <p>
+              This app embeds a Lens visualization by specifying the configuration. Data fetching
+              and rendering is completely managed by Lens itself.
+            </p>
+            <p>
+              The Change color button will update the configuration by picking a new random color of
+              the series which causes Lens to re-render. The Edit button will take the current
+              configuration and navigate to a prefilled editor.
+            </p>
+            <EuiFlexGroup>
+              <EuiFlexItem grow={false}>
+                <EuiButton
+                  onClick={() => {
+                    // eslint-disable-next-line no-bitwise
+                    const newColor = '#' + ((Math.random() * 0xffffff) << 0).toString(16);
+                    setColor(newColor);
+                  }}
+                >
+                  Change color
+                </EuiButton>
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                <EuiButton
+                  isDisabled={!props.plugins.lens.canUseEditor()}
+                  onClick={() => {
+                    props.plugins.lens.navigateToPrefilledEditor({
+                      id: '',
+                      timeRange: {
+                        from: 'now-5d',
+                        to: 'now',
+                      },
+                      attributes: getLensAttributes(
+                        props.core.uiSettings.get('defaultIndex'),
```

Comment:
So I ran into this by coincidence, but the `@timestamp` field isn't available on the ecommerce data, which causes the example to crash. Maybe the example should be tied to a specific index like logs?

Reply:
Good point, I fixed this by reading the index pattern and using the default time field